components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/TenantWorkflowConfigHolder.java

```diff
@@ -152,15 +188,29 @@ public void load() throws WorkflowException, RegistryException {
             workflowElem = workflowExtensionsElem.getFirstChildWithName(
                     new QName(WorkflowConstants.SUBSCRIPTION_DELETION));
             executorClass = workflowElem.getAttributeValue(new QName(WorkflowConstants.EXECUTOR));
-            clazz = TenantWorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            try {
+                clazz = TenantWorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
+            } catch (ClassNotFoundException e) {
+                clazz = TenantWorkflowConfigHolder.class.getClassLoader().loadClass(WorkflowConstants.
+                        DEFAULT_SUBSCRIPTION_DELETION);
+                workflowElem = AXIOMUtil.stringToOM(WorkflowConstants.
+                        DEFAULT_SUBSCRIPTION_DELETION_WORKFLOW_ELEMENT);
+            }
```

Comment:
You could do this much simpler way. See below sample.

        try {
            clazz = TenantWorkflowConfigHolder.class.getClassLoader().loadClass(executorClass);
            workFlowExecutor = (WorkflowExecutor) clazz.newInstance();
            loadProperties(workflowElem, workFlowExecutor);
        } catch (ClassNotFoundException e) {
            workFlowExecutor = new SubscriptionCreationSimpleWorkflowExecutor();
        }
        workflowExecutorMap.put(WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_CREATION, workFlowExecutor);


As you can see, you don't have to keep constants for the default one. 

Another this is, `loadProperties(workflowElem, workFlowExecutor);` is only needed if there are additional properties defined in the workflow configuration section. Default executor does not have any default configurations. so we could `loadProperties()` in to the try block. This prevents the need to have constants for  XML element

Reply:
Refactored